'ssize_t' does not name a type in ccnl-core.h:148:5 - CCNLite-Arduino

src/../../ccnl-core.h:148:5 error: 'ssize_t' doesn not name a type
ssize_t datalen;
I have included ethernet libraries successfully and upon running "make all upload" gives this error.
Can anyone tell me where i am going wrong here?

--- a/src/ccn-lite-arduino.c
+++ b/src/ccn-lite-arduino.c
@@ -234,6 +234,7 @@ extern struct ccnl_prefix_s sensor;
 int local_producer(struct ccnl_relay_s *relay, struct ccnl_face_s *from,
                    struct ccnl_pkt_s *pkt);
 
+#include "Time.h"
 #include "ccnl-core.c"
 #include "ccnl-ext-hmac.c"
 #include "ccnl-ext-http.c"

--- a/src/ccn-lite-arduino.c
+++ b/src/ccn-lite-arduino.c
@@ -137,6 +137,8 @@ struct sockaddr {
 #define AF_PACKET 1
 #define AF_INET   2
 
+#define	INADDR_BROADCAST	((unsigned long int) 0xffffffff)
+
 # define ntohs(i)  (((i<<8) & 0xff00) | ((i>>8) & 0x0ff))
 # define htons(i)  ntohs(i)
 

--- a/src/ccn-lite-arduino.c
+++ b/src/ccn-lite-arduino.c
@@ -153,6 +153,8 @@ ntohl(unsigned long l) {
 
 #define htonl(i) ntohl(i)
 
+typedef long ssize_t;
+
 // ----------------------------------------------------------------------
 
 #include <assert.h>

--- a/src/ccn-lite-arduino.c
+++ b/src/ccn-lite-arduino.c
@@ -157,6 +157,8 @@ ntohl(unsigned long l) {
 
 typedef long ssize_t;
 
+#define PRIi32  "li"
+
 // ----------------------------------------------------------------------
 
 #include <assert.h>

